### DISCUSSION TITLE
what does "cloud workload" mean?

### DISCUSSION THREAD

**Discussion by bobooscar** [cloud-hypervisor/cloud-hypervisor#3960] [General]
> “The project focuses on exclusively running modern, cloud workloads, on top of a limited set of hardware architectures and platforms. Cloud workloads refers to those that are usually run by customers inside a cloud provider. For our purposes this means modern operating systems with most I/O handled by paravirtualised devices (i.e. virtio), no requirement for legacy devices, and 64-bit CPUs.”

I'm wondering what does "cloud workload" mean here, does it mean "workloads in the guest that runs on public cloud, but not private cloud",  and , the guest OS is not built by the tenant by the cloud provider? so, the cloud provider could just provide any kind of network, disk, cpu, memory for tenants, that meets their requirement, and ,  as the tenants do not care what kind of network/disk, they just care about the performance/ reliability, etc. thus, we choose the cloud friendly devices such as virtio-XX , and leave legacy devices(e1000, etc.) out of our consideration?

in the issue: https://github.com/cloud-hypervisor/cloud-hypervisor/issues/3212
it says that display is not needed for cloud oriented workloads. after reading that, I'm getting more confused, why is that? on public cloud, people still needs display/vnc to make work conviently, eg, do word/excel office work, programming on eclipse,...  why is display not necessary for "cloud workload", what does "cloud workload"/"cloud oriented workload" mean exactly?

thanks!

**Comment by rbradford** [comment]
@bobooscar I agree we lack the terminology to effectively describe. By "cloud workloads" we are trying to encapsulate legacy free (or as close as we can achieve), headless workloads.

I accept that people do use virtual desktops in the cloud but we are not targetting that workload. This is from a view of pragmatism: the use of that feature is relatively niche and the complexity required to implement is would bloat the project: virtio-gpu, virtio-vga, virtio-input, virtio-sound + the RDP/VNC/SPICE infrastructure to make it accessible + authentication + TLS stack. This is a view that we have had from the origins of the project and are unlikely to deviate from.
